File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/MassStorageSample/src/com/hp/workpath/sample/massstoragesample/FileListAdapter.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.massstoragesample;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.hp.workpath.sample.massstoragesample.model.FileInfo;

import java.util.ArrayList;
import java.util.List;

public class FileListAdapter extends RecyclerView.Adapter<FileListAdapter.ViewHolder> implements View.OnClickListener {

    public static final String BACK = "<< (UP)";

    private final FileListOnClickListener mFileListOnClickListener;
    private List<FileInfo> mData = new ArrayList<>();
    private boolean[] mCheckedItemArray;

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView fileNameTextView;
        public CheckBox fileCheckBox;

        ViewHolder(View view) {
            super(view);
            fileNameTextView = view.findViewById(R.id.fileNameTextView);
            fileCheckBox = view.findViewById(R.id.checkBox);
        }
    }

    public interface FileListOnClickListener {
        void onDeleteButton(boolean enable);

        void onRenameButton(boolean enable);

        void onItemClick(View view);
    }

    public FileListAdapter(FileListOnClickListener fileListOnClickListener) {
        this.mFileListOnClickListener = fileListOnClickListener;
    }

    public void clear() {
        Log.i(MainActivity.TAG, "FileListAdapter clear");
        mCheckedItemArray = null;
        mData.clear();
        notifyItemRangeRemoved(0, mData.size());
    }

    public FileInfo getItem(int position) {
        return mData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_file, parent, false);
        view.setOnClickListener(this);
        view.findViewById(R.id.checkBox).setOnClickListener(this);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        FileInfo file = getItem(position);
        initViewHolder(holder);
        holder.fileCheckBox.setTag(position);
        holder.fileNameTextView.setText(file.getFileName());
        if (BACK.equals(file.getFileName())) {
            holder.fileCheckBox.setVisibility(View.INVISIBLE);
        }
        holder.fileCheckBox.setChecked(mCheckedItemArray[position]);
    }

    private void initViewHolder(ViewHolder holder) {
        if (holder != null) {
            holder.fileCheckBox.setVisibility(View.VISIBLE);
            holder.fileNameTextView.setText("");
        }
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.checkBox) {
            mCheckedItemArray[(int) v.getTag()] = ((CheckBox) v).isChecked();
            int checkedCount = 0;
            for (boolean b : mCheckedItemArray) {
                if (b) {
                    ++checkedCount;
                }
            }
            mFileListOnClickListener.onDeleteButton(checkedCount >= 1);
            mFileListOnClickListener.onRenameButton(checkedCount == 1);
        } else {
            mFileListOnClickListener.onItemClick(v);
        }
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    public boolean[] getCheckedItemArray() {
        return mCheckedItemArray;
    }

    public void setListItems(List<FileInfo> dataList) {
        mData = dataList;
        mCheckedItemArray = new boolean[mData.size()];
        notifyDataSetChanged();
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/MassStorageSample/src/com/hp/workpath/sample/massstoragesample/FileListAdapter.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.massstoragesample

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.hp.workpath.sample.massstoragesample.model.FileInfo

class FileListAdapter(private val mFileListOnClickListener: FileListOnClickListener) : RecyclerView.Adapter<FileListAdapter.ViewHolder>() {

    private var mData: MutableList<FileInfo> = ArrayList()
    private var checkedItemArray: BooleanArray? = null

    inner class ViewHolder(inflater: LayoutInflater, parent: ViewGroup) :
            RecyclerView.ViewHolder(inflater.inflate(R.layout.layout_file, parent, false)) {
        var fileNameTextView: TextView = itemView.findViewById(R.id.fileNameTextView)
        var fileCheckBox: CheckBox = itemView.findViewById(R.id.checkBox)

        init {
            itemView.setOnClickListener(clickListener)
            fileCheckBox.setOnClickListener(clickListener);
        }

        fun initViewHolder(position: Int) {
            fileCheckBox.tag = position
            fileCheckBox.visibility = View.VISIBLE
            fileNameTextView.text = ""
        }
    }

    interface FileListOnClickListener {
        fun onDeleteButton(enable: Boolean)
        fun onRenameButton(enable: Boolean)
        fun onItemClick(view: View)
    }

    fun clear() {
        Log.i(MainActivity.TAG, "FileListAdapter clear")
        checkedItemArray = null
        mData.clear()
        notifyItemRangeRemoved(0, mData.size)
    }

    fun getItem(position: Int): FileInfo {
        return mData[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int {
        return mData.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(inflater, parent)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val file = getItem(position)
        holder.initViewHolder(position)
        holder.fileCheckBox.tag = position
        holder.fileNameTextView.text = file.fileName
        if (BACK == file.fileName) {
            holder.fileCheckBox.visibility = View.INVISIBLE
        }
        checkedItemArray?.run {
            holder.fileCheckBox.isChecked = this[position]
        }
    }

    val clickListener: View.OnClickListener = View.OnClickListener { v ->
        if (v.id == R.id.checkBox) {
            checkedItemArray?.let {
                it[v.tag as Int] = (v as CheckBox).isChecked
                var checkedCount = 0
                for (x in it.indices) {
                    if (it[x]) {
                        ++checkedCount
                    }
                }
                mFileListOnClickListener.onDeleteButton(checkedCount >= 1)
                mFileListOnClickListener.onRenameButton(checkedCount == 1)
            }
        } else {
            mFileListOnClickListener.onItemClick(v)
        }
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    fun getCheckedItemArray(): BooleanArray? {
        return checkedItemArray
    }

    fun setListItems(data: MutableList<FileInfo>) {
        mData = data
        checkedItemArray = BooleanArray(data.size)
        notifyDataSetChanged()
    }

    companion object {
        const val BACK = "<< (UP)"
    }
}